packages/smooth_app/lib/pages/scan/ml_kit_scan_page.dart

```diff
@@ -0,0 +1,278 @@
+            scan_page_helper.areButtonsRendered(_model) ? 48 : 0;
+        final double availableScanHeight =
+            constraints.maxHeight - carouselHeight - buttonRowHeight;
+        // Padding for the qr code scanner. This ensures the scanner has equal spacing between buttons and carousel.
+        final EdgeInsets qrScannerPadding = EdgeInsets.only(
+            top: (availableScanHeight - scannerSize.height) / 2 +
+                buttonRowHeight);
+
+        return Stack(
+          fit: StackFit.expand,
+          children: <Widget>[
+            Container(
+              alignment: Alignment.center,
+              color: Colors.black,
+              child: Padding(
+                padding: qrScannerPadding,
+                child: SvgPicture.asset(
+                  'assets/actions/scanner_alt_2.svg',
+                  width: 60.0,
+                  height: 6,
+                  color: Colors.white,
+                ),
+              ),
+            ),
+            SmoothRevealAnimation(
+              delay: 400,
+              startOffset: Offset.zero,
+              animationCurve: Curves.easeInOutBack,
+              child: Transform.scale(
+                scale: scale,
+                child: Center(
+                  child: CameraPreview(_controller!),
+                ),
+              ),
+            ),
+            SmoothRevealAnimation(
+              delay: 400,
+              startOffset: const Offset(0.0, 0.1),
+              animationCurve: Curves.easeInOutBack,
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.start,
+                children: <Widget>[
+                  Padding(
+                    padding: qrScannerPadding,
+                    child: SmoothViewFinder(
+                      boxSize: scannerSize,
+                      lineLength: screenSize.width * 0.8,
+                    ),
+                  ),
+                ],
+              ),
+            ),
+            SmoothRevealAnimation(
+              delay: 400,
+              startOffset: const Offset(0.0, -0.1),
+              animationCurve: Curves.easeInOutBack,
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.start,
+                children: <Widget>[
+                  scan_page_helper.buildButtonsRow(context, _model),
+                  const Spacer(),
+                  SmoothProductCarousel(
+                    showSearchCard: true,
+                    height: carouselHeight,
+                  ),
+                ],
+              ),
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  Future<void> _startLiveFeed() async {
+    final CameraDescription camera = cameras[_cameraIndex];
+    _controller = CameraController(
+      camera,
+      ResolutionPreset.high,
+      enableAudio: false,
+    );
+    _controller?.initialize().then((_) {
+      if (!mounted) {
+        return;
+      }
+      _controller?.setFocusMode(FocusMode.auto);
```

Comment:
We can also set it to manual and allow for focusing on a certain focus offset as there is the option in the native android app, which is relevant for devices with poor autofocus
@teolemon